### PR TITLE
Issue #1 change project structure

### LINKED ISSUE #1: Change project structure
Change repo file structure to closely match common templates:

root:
- src
-- VolleyManagement.Backend
-- volley.org.ua_integration
-- VolleyManagement.Frontend
-- *.sln file
- build (move contents of BuildFiles here)
- tests (Tests for the backend)
- <remove .nuget folder>

AC:
- make sure build scripts, tests, solution build and angular build are working
- update CI for proper Cake build path (edit appveyor.yaml)

### PR DESCRIPTION
**Pull request by MikolaB**
closes #1

### CODE REVIEW

**Comment by sdiachen** [comment]
Great! Sonar works! :) We can ignore those issues by now - I just changed a bunch of rules

File: build/build.cake [hunk 1]
```diff
@@ -22,11 +22,11 @@ var sonarToken = HasArgument("sonar-token")
 var srcPath        = "./../";
 var srcDir         = Directory(srcPath);
 var buildDir       = srcDir + Directory("bin/") + Directory(configuration);
-var webBuildDir    = srcDir + Directory("VolleyManagement.UI/bin");
+var webBuildDir    = srcDir + Directory("src/VolleyManagement.Backend/VolleyManagement.UI/bin");
```
Comment (sdiachen) [inline build/build.cake]: why we added "src/" only here but not for buildDir?

File: src/VolleyManagement.Backend/VolleyManagement.Crosscutting/VolleyManagement.Crosscutting.csproj [hunk 2]
```diff
@@ -22,44 +22,46 @@
...
     </Reference>
     <Reference Include="Microsoft.Practices.EnterpriseLibrary.Logging, Version=6.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
       <SpecificVersion>False</SpecificVersion>
-      <HintPath>..\packages\EnterpriseLibrary.Logging.6.0.1304.0\lib\NET45\Microsoft.Practices.EnterpriseLibrary.Logging.dll</HintPath>
+      <HintPath>..\..\packages\EnterpriseLibrary.Logging.6.0.1304.0\lib\NET45\Microsoft.Practices.EnterpriseLibrary.Logging.dll</HintPath>
     </Reference>
     <Reference Include="System" />
     <Reference Include="System.Core" />
+    <Reference Include="System.Xml">
+      <HintPath>..\..\..\..\..\..\..\..\..\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7\System.Xml.dll</HintPath>
```
Comment (sdiachen) [inline src/VolleyManagement.Backend/VolleyManagement.Crosscutting/VolleyManagement.Crosscutting.csproj]: revert this change

Comment (MikolaB) [reply src/VolleyManagement.Backend/VolleyManagement.Crosscutting/VolleyManagement.Crosscutting.csproj]: Will do.

File: build/build.cake (hunk 1 above)
Comment (MikolaB) [reply build/build.cake]: Because path to build folder (folders with Debug/Release/UT dlls) is "rootFolder\bin" and for web build folder path is "rootFolder\src\VolleyManagement.Backend\VolleyManagement.UI\bin"

Comment (sdiachen) [reply build/build.cake]: then we need to rename "srcPath" to "rootPath" to better reveal intent. Adding "src" to srcPath variable seems strange

**Comment by sdiachen** [comment]
SonarQube analysis reported 166 issues
* ![CRITICAL][CRITICAL] 7 critical
* ![MAJOR][MAJOR] 64 major
* ![MINOR][MINOR] 92 minor
* ![INFO][INFO] 3 info

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [EnterpriseLibraryLog.cs#L30](https://github.com/VolleyManagement/volley-management/blob/90519612ac921950ce903d9e242b34ec2d5d29f3/src/VolleyManagement.Backend/VolleyManagement.Crosscutting/Logger/EnterpriseLibraryLog.cs#L30): Add a 'default' clause to this 'switch' statement. [rule](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS131)
1. ![CRITICAL][CRITICAL] [Score.cs#L21](https://github.com/VolleyManagement/volley-management/blob/90519612ac921950ce903d9e242b34ec2d5d29f3/src/VolleyManagement.Backend/VolleyManagement.Domain/GamesAggregate/Score.cs#L21): Use the overloading mechanism instead of the optional parameters. [rule](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS2360)
1. ![CRITICAL][CRITICAL] [SeedDataGenerator.cs#L777](https://github.com/VolleyManagement/volley-management/blob/90519612ac921950ce903d9e242b34ec2d5d29f3/src/VolleyManagement.Backend/VolleyManagement.Data.MsSql/Context/Migrations/SeedDataGenerator.cs#L777): The Cyclomatic Complexity of this method is 17 which is greater than 10 authorized. [rule](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS1541)
1. ![CRITICAL][CRITICAL] [SeedDataGenerator.cs#L911](https://github.com/VolleyManagement/volley-management/blob/90519612ac921950ce903d9e242b34ec2d5d29f3/src/VolleyManagement.Backend/VolleyManagement.Data.MsSql/Context/Migrations/SeedDataGenerator.cs#L911): The Cyclomatic Complexity of this method is 14 which is greater than 10 authorized. [rule](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS1541)
1. ![CRITICAL][CRITICAL] [SeedDataGenerator.cs#L966](https://github.com/VolleyManagement/volley-management/blob/90519612ac921950ce903d9e242b34ec2d5d29f3/src/VolleyManagement.Backend/VolleyManagement.Data.MsSql/Context/Migrations/SeedDataGenerator.cs#L966): Refactor this code to not nest more than 3 control flow statements. [rule](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS134)
1. ![CRITICAL][CRITICAL] [SeedDataGenerator.cs#L976](https://github.com/VolleyManagement/volley-management/blob/90519612ac921950ce903d9e242b34ec2d5d29f3/src/VolleyManagement.Backend/VolleyManagement.Data.MsSql/Context/Migrations/SeedDataGenerator.cs#L976): Refactor this code to not nest more than 3 control flow statements. [rule](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS134)
1. ![CRITICAL][CRITICAL] [SeedDataGenerator.cs#L1057](https://github.com/VolleyManagement/volley-management/blob/90519612ac921950ce903d9e242b34ec2d5d29f3/src/VolleyManagement.Backend/VolleyManagement.Data.MsSql/Context/Migrations/SeedDataGenerator.cs#L1057): The Cyclomatic Complexity of this method is 15 which is greater than 10 authorized. [rule](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS1541)
1. ![MAJOR][MAJOR] [UserModel.cs#L48](https://github.com/VolleyManagement/volley-management/blob/90519612ac921950ce903d9e242b34ec2d5d29f3/src/VolleyManagement.Backend/VolleyManagement.Contracts/Authentication/Models/UserModel.cs#L48): Refactor this property to use a generic collection designed for inheritance. [rule](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS3956)
1. ![MAJOR][MAJOR] [AllowedOperations.cs#L19](https://github.com/VolleyManagement/volley-management/blob/90519612ac921950ce903d9e242b34ec2d5d29f3/src/VolleyManagement.Backend/VolleyManagement.Contracts/Authorization/AllowedOperations.cs#L19): Refactor this constructor to use a generic collection designed for inheritance. [rule](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS3956)
1. ![MAJOR][MAJOR] [IAuthorizationService.cs#L24](https://github.com/VolleyManagement/volley-management/blob/90519612ac921950ce903d9e242b34ec2d5d29f3/src/VolleyManagement.Backend/VolleyManagement.Contracts/Authorization/IAuthorizationService.cs#L24): Refactor this method to use a generic collection designed for inheritance. [rule](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS3956)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: src/VolleyManagement.Backend/VolleyManagement.Data.MsSql/VolleyManagement.Data.MsSql.csproj [hunk 3]
```diff
@@ -251,6 +251,10 @@
     <Compile Include="Context\VolleyManagementEntities.cs" />
   </ItemGroup>
   <ItemGroup>
+    <ProjectReference Include="..\VolleyManagement.Contracts\VolleyManagement.Contracts.csproj">
```
Comment (sdiachen) [inline src/VolleyManagement.Backend/VolleyManagement.Data.MsSql/VolleyManagement.Data.MsSql.csproj]: Why do you need this?

Comment (MikolaB) [reply src/VolleyManagement.Backend/VolleyManagement.Data.MsSql/VolleyManagement.Data.MsSql.csproj]: My bad. Added by mistake during structure change. Will remove.

**Comment by sdiachen** [comment]
SonarQube analysis reported 166 issues
* ![CRITICAL][CRITICAL] 7 critical
* ![MAJOR][MAJOR] 64 major
* ![MINOR][MINOR] 92 minor
* ![INFO][INFO] 3 info

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [EnterpriseLibraryLog.cs#L30](https://github.com/VolleyManagement/volley-management/blob/fbe6252e9ec0f0ed18bc7711650ee81c48ea6af5/src/VolleyManagement.Backend/VolleyManagement.Crosscutting/Logger/EnterpriseLibraryLog.cs#L30): Add a 'default' clause to this 'switch' statement. [rule](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS131)
1. ![CRITICAL][CRITICAL] [Score.cs#L21](https://github.com/VolleyManagement/volley-management/blob/fbe6252e9ec0f0ed18bc7711650ee81c48ea6af5/src/VolleyManagement.Backend/VolleyManagement.Domain/GamesAggregate/Score.cs#L21): Use the overloading mechanism instead of the optional parameters. [rule](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS2360)
1. ![CRITICAL][CRITICAL] [SeedDataGenerator.cs#L777](https://github.com/VolleyManagement/volley-management/blob/fbe6252e9ec0f0ed18bc7711650ee81c48ea6af5/src/VolleyManagement.Backend/VolleyManagement.Data.MsSql/Context/Migrations/SeedDataGenerator.cs#L777): The Cyclomatic Complexity of this method is 17 which is greater than 10 authorized. [rule](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS1541)
1. ![CRITICAL][CRITICAL] [SeedDataGenerator.cs#L911](https://github.com/VolleyManagement/volley-management/blob/fbe6252e9ec0f0ed18bc7711650ee81c48ea6af5/src/VolleyManagement.Backend/VolleyManagement.Data.MsSql/Context/Migrations/SeedDataGenerator.cs#L911): The Cyclomatic Complexity of this method is 14 which is greater than 10 authorized. [rule](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS1541)
1. ![CRITICAL][CRITICAL] [SeedDataGenerator.cs#L966](https://github.com/VolleyManagement/volley-management/blob/fbe6252e9ec0f0ed18bc7711650ee81c48ea6af5/src/VolleyManagement.Backend/VolleyManagement.Data.MsSql/Context/Migrations/SeedDataGenerator.cs#L966): Refactor this code to not nest more than 3 control flow statements. [rule](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS134)
1. ![CRITICAL][CRITICAL] [SeedDataGenerator.cs#L976](https://github.com/VolleyManagement/volley-management/blob/fbe6252e9ec0f0ed18bc7711650ee81c48ea6af5/src/VolleyManagement.Backend/VolleyManagement.Data.MsSql/Context/Migrations/SeedDataGenerator.cs#L976): Refactor this code to not nest more than 3 control flow statements. [rule](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS134)
1. ![CRITICAL][CRITICAL] [SeedDataGenerator.cs#L1057](https://github.com/VolleyManagement/volley-management/blob/fbe6252e9ec0f0ed18bc7711650ee81c48ea6af5/src/VolleyManagement.Backend/VolleyManagement.Data.MsSql/Context/Migrations/SeedDataGenerator.cs#L1057): The Cyclomatic Complexity of this method is 15 which is greater than 10 authorized. [rule](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS1541)
1. ![MAJOR][MAJOR] [UserModel.cs#L48](https://github.com/VolleyManagement/volley-management/blob/fbe6252e9ec0f0ed18bc7711650ee81c48ea6af5/src/VolleyManagement.Backend/VolleyManagement.Contracts/Authentication/Models/UserModel.cs#L48): Refactor this property to use a generic collection designed for inheritance. [rule](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS3956)
1. ![MAJOR][MAJOR] [AllowedOperations.cs#L19](https://github.com/VolleyManagement/volley-management/blob/fbe6252e9ec0f0ed18bc7711650ee81c48ea6af5/src/VolleyManagement.Backend/VolleyManagement.Contracts/Authorization/AllowedOperations.cs#L19): Refactor this constructor to use a generic collection designed for inheritance. [rule](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS3956)
1. ![MAJOR][MAJOR] [IAuthorizationService.cs#L24](https://github.com/VolleyManagement/volley-management/blob/fbe6252e9ec0f0ed18bc7711650ee81c48ea6af5/src/VolleyManagement.Backend/VolleyManagement.Contracts/Authorization/IAuthorizationService.cs#L24): Refactor this method to use a generic collection designed for inheritance. [rule](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS3956)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by sdiachen** [comment]
SonarQube analysis reported 166 issues
* ![CRITICAL][CRITICAL] 7 critical
* ![MAJOR][MAJOR] 64 major
* ![MINOR][MINOR] 92 minor
* ![INFO][INFO] 3 info

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [EnterpriseLibraryLog.cs#L30](https://github.com/VolleyManagement/volley-management/blob/1d545ee27bb432db0873baa2988af2508dd85212/src/VolleyManagement.Backend/VolleyManagement.Crosscutting/Logger/EnterpriseLibraryLog.cs#L30): Add a 'default' clause to this 'switch' statement. [rule](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS131)
1. ![CRITICAL][CRITICAL] [Score.cs#L21](https://github.com/VolleyManagement/volley-management/blob/1d545ee27bb432db0873baa2988af2508dd85212/src/VolleyManagement.Backend/VolleyManagement.Domain/GamesAggregate/Score.cs#L21): Use the overloading mechanism instead of the optional parameters. [rule](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS2360)
1. ![CRITICAL][CRITICAL] [SeedDataGenerator.cs#L777](https://github.com/VolleyManagement/volley-management/blob/1d545ee27bb432db0873baa2988af2508dd85212/src/VolleyManagement.Backend/VolleyManagement.Data.MsSql/Context/Migrations/SeedDataGenerator.cs#L777): The Cyclomatic Complexity of this method is 17 which is greater than 10 authorized. [rule](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS1541)
1. ![CRITICAL][CRITICAL] [SeedDataGenerator.cs#L911](https://github.com/VolleyManagement/volley-management/blob/1d545ee27bb432db0873baa2988af2508dd85212/src/VolleyManagement.Backend/VolleyManagement.Data.MsSql/Context/Migrations/SeedDataGenerator.cs#L911): The Cyclomatic Complexity of this method is 14 which is greater than 10 authorized. [rule](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS1541)
1. ![CRITICAL][CRITICAL] [SeedDataGenerator.cs#L966](https://github.com/VolleyManagement/volley-management/blob/1d545ee27bb432db0873baa2988af2508dd85212/src/VolleyManagement.Backend/VolleyManagement.Data.MsSql/Context/Migrations/SeedDataGenerator.cs#L966): Refactor this code to not nest more than 3 control flow statements. [rule](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS134)
1. ![CRITICAL][CRITICAL] [SeedDataGenerator.cs#L976](https://github.com/VolleyManagement/volley-management/blob/1d545ee27bb432db0873baa2988af2508dd85212/src/VolleyManagement.Backend/VolleyManagement.Data.MsSql/Context/Migrations/SeedDataGenerator.cs#L976): Refactor this code to not nest more than 3 control flow statements. [rule](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS134)
1. ![CRITICAL][CRITICAL] [SeedDataGenerator.cs#L1057](https://github.com/VolleyManagement/volley-management/blob/1d545ee27bb432db0873baa2988af2508dd85212/src/VolleyManagement.Backend/VolleyManagement.Data.MsSql/Context/Migrations/SeedDataGenerator.cs#L1057): The Cyclomatic Complexity of this method is 15 which is greater than 10 authorized. [rule](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS1541)
1. ![MAJOR][MAJOR] [UserModel.cs#L48](https://github.com/VolleyManagement/volley-management/blob/1d545ee27bb432db0873baa2988af2508dd85212/src/VolleyManagement.Backend/VolleyManagement.Contracts/Authentication/Models/UserModel.cs#L48): Refactor this property to use a generic collection designed for inheritance. [rule](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS3956)
1. ![MAJOR][MAJOR] [AllowedOperations.cs#L19](https://github.com/VolleyManagement/volley-management/blob/1d545ee27bb432db0873baa2988af2508dd85212/src/VolleyManagement.Backend/VolleyManagement.Contracts/Authorization/AllowedOperations.cs#L19): Refactor this constructor to use a generic collection designed for inheritance. [rule](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS3956)
1. ![MAJOR][MAJOR] [IAuthorizationService.cs#L24](https://github.com/VolleyManagement/volley-management/blob/1d545ee27bb432db0873baa2988af2508dd85212/src/VolleyManagement.Backend/VolleyManagement.Contracts/Authorization/IAuthorizationService.cs#L24): Refactor this method to use a generic collection designed for inheritance. [rule](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS3956)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'
